src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+        at.y = eye.y - 90.0f;
+        at.z = eye.z - (Math_CosS(chest->yawTowardsPlayer) * 250.0f);
+        Play_SetCameraAtEye(play, CamId, &at, &eye);
+    } else if (this->timer < 145) {
+        func_80ADF4E0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADEDF8.s")
+void func_80ADF4E0(EnTakaraya* this) {
+    Interface_StartTimer(TIMER_ID_MINIGAME_2, 45);
+    this->timer = 1;
+    this->actionFunc = &func_80ADF520;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADEE4C.s")
+void func_80ADF520(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Play_InCsMode(play) == 0) {
+        if (Flags_GetTreasure(play, this->actor.params)) {
+            Flags_SetSwitch(play, this->unk2B0);
+            play->actorCtx.sceneFlags.chest &= ~(1 << this->actor.params);
+            this->timer = 0;
+            gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_6;
+            func_80ADF608(this, play);
+        } else if (gSaveContext.timerCurTimes[TIMER_ID_MINIGAME_2] == TIMER_STATE_OFF) {
+            this->timer = 50;
+            Message_StartTextbox(play, 0x77D, &this->actor);
+            gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+            func_80ADF608(this, play);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADEF74.s")
+void func_80ADF608(EnTakaraya* this, PlayState* play) {
+    func_800B7298(play, &this->actor, PLAYER_CSMODE_WAIT);
+    this->unk2AD = true;
+    this->actionFunc = &func_80ADF654;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF03C.s")
+void func_80ADF654(EnTakaraya* this, PlayState* play) {
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        if (gSaveContext.timerStates[TIMER_ID_MINIGAME_2] == TIMER_STATE_6) {
+            play->transitionType = TRANS_TYPE_80;
+            gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE;
+        } else {
+            play->transitionType = TRANS_TYPE_64;
+            gSaveContext.nextTransitionType = TRANS_TYPE_FADE_BLACK;
+        }
+        gSaveContext.nextCutsceneIndex = 0;
+        play->nextEntrance = ENTRANCE(TREASURE_CHEST_SHOP, 1);
+        play->transitionTrigger = TRANS_TRIGGER_START;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF050.s")
+void func_80ADF6DC(EnTakaraya* this) {
+    Animation_PlayLoop(&this->skelAnime, &object_bg_Anim_001384);
+    this->eyeTexIndex = 0;
+    this->actor.flags |= ACTOR_FLAG_10000;
+    this->actionFunc = &func_80ADF730;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF2D4.s")
+void func_80ADF730(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actor.flags &= 0xFFFEFFFF;
```

Comment:
```suggestion
        this->actor.flags &= ~ACTOR_FLAG_10000;
```
if you inverse 0xFFFEFFFF you get 0x00010000, so this is a clearing that one flag.